pyQuARC/code/downloader.py

```diff
@@ -91,24 +92,32 @@ def download(self):
 
         # is the concept id valid? if not, log error
         if not self._valid_concept_id():
-            self.log_error(
-                "invalid_concept_id",
-                {"concept_id": self.concept_id}
-            )
+            self.log_error("invalid_concept_id", {"concept_id": self.concept_id})
             return
 
         # constructs url based on concept id
         url = self._construct_url()
         headers = get_headers()
         response = requests.get(url, headers=headers)
+
+        # if the authorization token is invalid, even public metadata that doesn't require the token is inaccessible
+        # this works around that
+        if response.status_code == 401:  # if token invalid, try without token
+            response = requests.get(url)
+
         # gets the response, makes sure it's 200, puts it in an object variable
         if response.status_code != 200:
+            try:
+                message = json.loads(response.text).get("errors")
+            except (json.decoder.JSONDecodeError, KeyError):
+                message = "Something went wrong while downloading the requested metadata. Make sure all the inputs are correct."
             self.log_error(
                 "request_failed",
                 {
                     "concept_id": self.concept_id,
                     "url": url,
                     "status_code": response.status_code,
+                    "message": message,
```

Comment:
If the error is related to invalid token, rerun request without header and see if that solves the problem
propagate the message from the cmr response in case of download failure